fastlane_bot/bot.py

```diff
@@ -916,6 +916,19 @@ def _handle_trade_instructions(
             best_trade_instructions,
         ) = r
 
+        indices = [
```

Comment:
my main concern is that it's a late phase and I would expect us to be able to not include these tokens/pools in the data we pass to the optimizer

Reply:
From what I've observed so far (as well as what I know), most of the execution time is spent on RPCs and not on purely internal computations (even the complicated ones taking place inside the optimizer).

Of course, this is not to say that we shouldn't strive to place this filter as early as possible within the process.

And regardless, the current solution is by itself most likely flawed, because as explained in[this comment of mine](https://github.com/bancorprotocol/fastlane-bot/pull/528/files#r1557935564) above, it might result with `best_profit` and `best_src_token` being "stale" values, referring to a token which has just been filtered out of those 3 trade-instruction lists.

So I'm now thinking of doing it function `get_curves`, immediately after the line:
```
tokens = self.db.get_tokens()
```